### DISCUSSION TITLE
How do I remove a matrix from an ArchR Project?

### DISCUSSION THREAD

**Discussion by markphillippebworth** [GreenleafLab/ArchR#843] [Feature Requests]
Hello!

I've been hitting errors with subsetting, and I've tracked it down to a matrix within my archr project. It somehow got duplicated, so now I have two matrices labeled "MarkerSet". How do I remove them from the ArchR Project /arrow file?

Copying over the arrows manually doesn't seem to help.

Thank you!

**Comment by rcorces** [comment]
We do not have a function that does this but it would be good to try to add. You would have to manually edit the HDF5 files and this seems a bit dangerous to me.

**Reply by markphillippebworth** [reply]
Yes... that would be really nice. Seurat has a functionality called DietSeurat. Something analoguous would be really nice for ArchR. 

I'm not quite up to manually editing the HDF5 files myself....

**Reply by rcorces** [reply]
Turns out ArchR has a hidden function that was originally designed to do this. It is not documented and you would need to know the name of the HDF5 group that you are trying to drop. Theoretically, this could be used. We have very limited bandwidth these days but we should make a wrapper around this function that works on the project level to remove a matrix from all arrows.

https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/ArrowUtils.R#L231

**Reply by markphillippebworth** [reply]
That's great! Thank you for digging that up.

**Reply by dagarfield** [reply]
@markphillippebworth , did you ever end up writing a function for this? I "solved" this before by exporting the fragment file and cellCol data and starting over, but I'd be curious for a more efficient method....

**Reply by markphillippebworth** [reply]
@dagarfield  I did, though it's been a while since I've needed to use it. Here's a public repo with the code, which is largely just re-purposed from internal ArchR data. 

https://github.com/aifimmunology/DietArchR
